packages/svelte/src/compiler/phases/2-analyze/visitors/CallExpression.js

```diff
@@ -179,6 +180,7 @@ export function CallExpression(node, context) {
 			context.state.expression.has_call = true;
 			context.state.expression.has_state = true;
 			context.state.expression.can_inline = false;
+			mark_subtree_dynamic(context.path);
```

Comment:
I think this is good, just for the sake of it, can we check wherever else we set `can_inline` to false to see if the subtree is actually marked dynamic?

Reply:
I added it to other places but didn't notice any differences. Which was odd, but this whole inline/dynamic system feels a bit wonky tbh.

Reply:
Yeah i think we need to restructure this a bit to keep a single source of truth (but i think it will be difficult)